pkg/api/customization/project/project_store.go

```diff
@@ -171,12 +175,51 @@ func isQuotaFit(apiContext *types.APIContext, nsQuotaLimit *v3.ResourceQuotaLimi
 		return err
 	}
 
-	if project.ResourceQuota == nil || project.ResourceQuota.UsedLimit == nil {
+	if project.ResourceQuota == nil {
 		return nil
 	}
 
+	// check if fields were added or removed
+	// and update project's namespaces accordingly
+	defaultQuotaLimitMap, err := convert.EncodeToMap(nsQuotaLimit)
+	if err != nil {
+		return err
+	}
+
+	usedQuotaLimitMap := map[string]interface{}{}
+	if project.ResourceQuota.UsedLimit != nil {
+		usedQuotaLimitMap, err = convert.EncodeToMap(project.ResourceQuota.UsedLimit)
+		if err != nil {
+			return err
+		}
+	}
+
+	limitToAdd := map[string]interface{}{}
+	limitToRemove := map[string]interface{}{}
+	for key, value := range defaultQuotaLimitMap {
+		if _, ok := usedQuotaLimitMap[key]; !ok {
+			limitToAdd[key] = value
```

Comment:
To understand - `defaultQuotaLimitMap` is the one corresponding to namespace, and `usedQuotaLimitMap ` is for project. So, if something is in namespace map, but not in project, should be in `limitToRemove`?

Reply:
correct. We don't want to check the field's limit agains the used limit if it's no longer in the `defaultQuotaLimitMap`